plugins/woocommerce/client/blocks/assets/js/blocks/checkout/inner-blocks/checkout-shipping-address-block/customer-address.tsx

```diff
@@ -27,24 +27,28 @@ const CustomerAddress = () => {
 	const { dispatchCheckoutEvent } = useStoreEvents();
 
 	// Forces editing state if store has errors.
-	const { hasValidationErrors, invalidProps } = useSelect(
+	const { hasValidationErrors, getValidationErrorSelector } = useSelect(
 		( select ) => {
 			const store = select( validationStore );
 			return {
 				hasValidationErrors: store.hasValidationErrors(),
-				invalidProps: Object.keys( shippingAddress )
-					.filter( ( key ) => {
-						return (
-							store.getValidationError( 'shipping_' + key ) !==
-							undefined
-						);
-					} )
-					.filter( Boolean ),
+				getValidationErrorSelector: store.getValidationError,
 			};
 		},
-		[ shippingAddress ]
+		[]
 	);
 
+	const invalidProps = useMemo( () => {
+		return Object.keys( shippingAddress )
+			.filter( ( key ) => {
+				return (
+					getValidationErrorSelector( 'shipping_' + key ) !==
+					undefined
+				);
+			} )
+			.filter( Boolean );
+	}, [ JSON.stringify( shippingAddress ), getValidationErrorSelector ] );
```

Comment:
Tested with and without stringify, did not notice any difference in amount of times the hook was called. Also tested with this snippet:

```js
const shippingAddressRef = useRef( shippingAddress );

useEffect( () => {
	console.log( 'Previous address:', shippingAddressRef );
	console.log( 'Current address:', shippingAddress );
	console.log(
		'Same reference?',
		shippingAddressRef.current === shippingAddress
	);

	// Update the ref after logging
	shippingAddressRef.current = shippingAddress;
}, [ shippingAddress ] );
```

and it's the same ref. The ref would not change unless the shipping address values change, in which case a recomputation of the memo is expected and OK.